pkg/webhook/server.go

```diff
@@ -246,6 +250,15 @@ func (s *DefaultServer) Start(ctx context.Context) error {
 	idleConnsClosed := make(chan struct{})
 	go func() {
 		<-ctx.Done()
+
+		// Disable HTTP keep-alives to close persistent connections after next
+		// HTTP request. Clients may reconnect until routes are updated and
+		// server listeners are closed however this should start gradually
+		// migrating clients to server instances that are not about to shutdown
+		srv.SetKeepAlivesEnabled(false)
+		// Wait before shutting down webhook server
+		time.Sleep(s.Options.ShutdownDelay)
```

Comment:
Why is this needed? Reading the intent of the PR, there is a context with a timeout that will be given to `shutdown` which allows those connections to be closed, was that not enough?